modules/internal/StringCasts.chpl

```diff
@@ -46,13 +46,13 @@ module StringCasts {
   proc _cast(type t, x: string) throws where isBoolType(t) {
     var str = x.strip();
     if str.isEmptyString() {
-      throw new IllegalArgumentError("bad cast from empty string to bool");
+      throw new unmanaged IllegalArgumentError("bad cast from empty string to bool");
```

Comment:
Geez, it is really unfortunate that the users will need to say what kind of error they are allocating for a "throw". I wish we could allow simply `throw new MyErrorClass(...)` and have the compiler figure out whether it should be unmanaged or what.

Reply:
I moved this to issue #9698.